internal/controllers/machine/machine_controller.go

```diff
@@ -189,10 +189,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Re
 			m.Spec.ClusterName, m.Name, m.Namespace)
 	}
 
+	s := &scope{
+		cluster: cluster,
+		machine: m,
+	}
+
 	// Return early if the object or Cluster is paused.
 	if annotations.IsPaused(cluster, m) {
 		log.Info("Reconciliation is paused for this object")
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, setPausedCondition(ctx, r.Client, s)
```

Comment:
Note: this does not work together with the above predicate because a paused object only eventually enters this code here (e.g. only when paused is set on the cluster object).